R/fwrite.R

```diff
@@ -103,6 +104,10 @@ fwrite = function(x, file="", append=FALSE, quote="auto",
       header=col.names, sep=sep, sep2=sep2, eol=eol, na.strings=na,
       dec=dec, qmethod=qmethod, logical01=logical01
     )
+    if (is.list(yaml)) {
+      yaml_header <- c(yaml_header, yaml)  
```

Comment:
(open-ended) is there any extra checks we should do on the `data.table` side here of the user input? or will it be fine to leave it up to `yaml::as.yaml` to do all validation?

Reply:
That was my thought, yes.